lib/p2p/packets/types/DisconnectingPacket.ts

```diff
@@ -6,6 +6,7 @@ import { removeUndefinedProps } from '../../../utils/utils';
 
 export type DisconnectingPacketBody = {
   reason: DisconnectionReason;
+  peerPubKey: string;
```

Comment:
We shouldn't have to add this to the disconnecting packet, particularly since this is a a breaking change. The peer should already know our pub key, and should use the previously known pubkey rather than the value from this packet.

Plus I figure this could introduce some weird vulnerability where a peer sends a disconnecting packet with another node's pub key.

Reply:
I suppose `expectedNodePubKey` is the peer's pub key ?

Reply:
Or the `_nodePubKey` in cases where a peer bans us after we have connected and initialized the session, I figure.